.gitlab-ci.yml

```diff
@@ -195,6 +195,57 @@ build-idp-image:
       --build-arg "LARGE_FILES_USER=${LARGE_FILES_USER}"
       --build-arg "SERVICE_PROVIDERS_KEY=${SERVICE_PROVIDERS_KEY}"
 
+build-nginx-image:
+  stage: review
+  needs: []
+  interruptible: true
+  variables:
+    BRANCH_TAGGING_STRING: ''
+  rules:
+    - if: $CI_COMMIT_BRANCH == $CI_DEFAULT_BRANCH
+      variables:
+        BRANCH_TAGGING_STRING: '--destination ${ECR_REGISTRY}/identity-idp/nginx:main'
+    - if: $CI_COMMIT_BRANCH != $CI_DEFAULT_BRANCH
```

Comment:
Can we add a condition to not run the build on a PR if the files haven't changed? It would still build on `main` right?

Reply:
wouldn't that break reviewapps if a given idp image lacks a matching nginx image with the same tag?

Reply:
Right, it would still get built on any change to `main`, which would be frequently enough for that?

Reply:
the image tag for idp and nginx need to match exactly or the current deploy process will fail, not really a guarantee that the images exist otherwise or to know what the "current" is

Reply:
Oh, I see what you mean.  Sorry, I thought you were saying only build on changes to the files while on main.  I'm pretty sure that we can rig that up.  

But it seems like we would want to build an nginx image on PRs too, in case we make changes to the dockerfile or .conf, right?

Reply:
Maybe I should ask:  is there a reason why we wouldn't want to build an image almost every time?

Reply:
I think avoiding repetitive image builds where possible gives us more room to work with on build times as we go forward. And yeah, for this suggestion, any file that is related to the image that changed should cause a build to be run.

I don't think it's a blocker, and the image build time is not significant currently, but much easier to have in place from the start.

Reply:
Yeah, this image build is never going to be the long pole here.  I do think there is value in it being built outside of main, since we will want to test nginx changes out in a lower environment before rolling it out, probably.

How about we land this and then we can fiddle with the rules later on?  I hate fiddling with rules.  I'd rather everything be done every time, so that there's no jobs that run randomly when I don't expect because I messed up the rule in some way.  :-)